### PR TITLE
[BallerinaToOpenAPI] Fix Invalid response code generated when adding return type annotation

### PR DESCRIPTION
**Pull request by lnash94**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIResponseMapper.java [hunk 1]
```diff
@@ -198,19 +206,77 @@ private ApiResponses extractReturnAnnotationDetails(OperationAdaptor operationAd
...
+                             *  resource function get pet02() returns @http:Payload {mediaType: "application/fake+xml"}
+                             *  User {
+                             *         return {};
+                             *     }
+                             *  }
+                             */
+                            StringBuilder stringBuilder = new StringBuilder();
+                            String[] splits = mediaType.split(SLASH);
+                            stringBuilder.append(splits[0]).append(SLASH).append(customMediaType.get())
+                                    .append("+").append(splits[1]);
```
Comment (NipunaRanasinghe) [inline openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIResponseMapper.java:267]: Shall we replace the hardcoded string with the common constant we have?

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java [hunk 2]
```diff
@@ -223,22 +229,80 @@ public static String getValidName(String identifier) {
...
+            List<String> expressionNode = extractAnnotationFieldDetails(annotationReference, annotationField,
+                    annotation);
+            if (!expressionNode.isEmpty()) {
+                return Optional.of(expressionNode.get(0));
+            }
+        }
+        return Optional.empty();
+    }
+
+    public static List<String> extractAnnotationFieldDetails(String annotationReference, String annotationField,
```
Comment (NipunaRanasinghe) [inline openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java:241]: Shall we add java docs for the new util method?

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java [hunk 3]
```diff
@@ -223,22 +229,80 @@ public static String getValidName(String identifier) {
...
+                } else {
+                    mediaTypes.add(expressionNode.toString().trim().replaceAll("\"", ""));
                 }
             }
         }
-        return Optional.empty();
+        return mediaTypes;
     }
 
+    public static List<String> extractAnnotationFieldDetails(String annotationReference, String annotationField,
```
Comment (NipunaRanasinghe) [inline openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java:270]: Why are we having overloaded util methods with duplicate logic? It seems to me that we can merge these two util methods.

File: openapi-cli/src/test/java/io/ballerina/openapi/generators/openapi/ModuleReferenceTests.java [hunk 4]
```diff
@@ -59,4 +59,10 @@ public void testResponse05() throws IOException {
         Path ballerinaFilePath = RES_DIR.resolve("arrayTypeResponse.bal");
         TestUtils.compareWithGeneratedFile(ballerinaFilePath, "arrayTypeResponse.yaml");
     }
+
+    @Test(description = "Response has payload annotation with media type define in the separate module.")
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/java/io/ballerina/openapi/generators/openapi/ModuleReferenceTests.java:63]:
```suggestion
    @Test(description = "Response has payload annotation with media type defined in a separate module.")
```

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java (hunk 3 above)
Comment (lnash94) [reply openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java:250]: Yeah merged it :)

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**